### DISCUSSION TITLE
DeliverySubject as a prefix to original subject name

### DISCUSSION THREAD

**Discussion by ealeykin** [nats-io/nats-server#4729] [Q&A] [answered]
Hi

I would like to use a push subscription/consumer with DeliverySubject. Looks like it expects an explicit and complete subject name, so if I consume from a stream containing multiple subject - they will all go to that single delivery subject. Im not using filter and I want them all to be pushed to a subject with a prefix + original subject name.

Is it possible to get original subject name from within delivery subject configuration or in subscription handler?

Thanks
Eugene

**Comment by derekcollison** [comment, marked as answer]
Even though the delivery subject is used for delivery (pull consumers also use different subjects to deliver the messages), the origin subject is presented in the callback or via the methods on the message itself. So you have that preserved and available.

**Comment by ealeykin** [comment]
Awesome, just checked that in the code and it works like you said! Was trying to find that somewhere in the documentation and failed.

Thank you for a quick reply!
